### ISSUE TITLE
[forwardport] don't error out on missing template

### ISSUE THREAD

**Issue by kinarashah** [rancher/rancher#24133]
https://github.com/rancher/rancher/pull/23928

**Comment by kinarashah** [comment]
https://github.com/rancher/rke/issues/1050

**Comment by jiaqiluo** [comment]
before this fix, every addon must support all k8s versions, and the validation test on the metadata checks for the range of k8s versions for each addon.

This brings a problem that the validation fails if a newly introduced addon supports only the newer k8s version not all versions.

how to test:
-  add a new addon like how nodeLocalDns is added 
- generate metadata, no error should show up 
- rancher logs shouldn't have any error on using this new metadata
 
We will wait for the this issue https://github.com/rancher/rke/issues/1050 and this PR  https://github.com/rancher/kontainer-driver-metadata/pull/51#discussion_r325212798  to be merged to validate this fix.

**Comment by jiaqiluo** [comment]
The fix is validated in rancher:master-head ff24cb9e3

Steps:
- add a new addons template `Nodelocal` wich only supports k8s version v1.17.0+
- generate the data.json file and push to a new branch  
- set Rancher server to enable debug logging
- set Rancher server to use that metadata

Results:
- Rancher log does not error out on the missing k8s versions, instead, the following warnings show in the log
```
2020/02/13 23:53:38 [INFO] driverMetadata: refresh data
2020/02/13 23:53:38 [DEBUG] getPluginData: no template found for k8sVersion v1.13.12-rancher2-1 plugin nodelocal
2020/02/13 23:53:38 [DEBUG] getPluginData: no template found for k8sVersion v1.14.7-rancher1-1 plugin nodelocal
2020/02/13 23:53:38 [DEBUG] getPluginData: no template found for k8sVersion v1.14.8-rancher2-1 plugin nodelocal
...
...
```
